app/helpers/field_slips_helper.rb

```diff
@@ -6,4 +6,42 @@ def last_observation
 
     ObservationView.last(User.current)
   end
+
+  def field_slip_show_tabs(field_slip)
+    links = [field_slips_index_tab, new_field_slip_tab]
+    return links unless field_slip.can_edit?
+
+    links.push(edit_field_slip_tab(field_slip),
+               destroy_field_slip_tab(field_slip))
+  end
+
+  def field_slip_edit_tabs(field_slip)
+    [field_slips_index_tab, show_field_slip_tab(field_slip)]
+  end
+
+  def field_slips_index_tabs
+    [new_field_slip_tab]
+  end
+
+  def field_slips_index_tab
+    [:field_slip_index.t, field_slips_path, { class: tab_id(__method__.to_s) }]
+  end
+
+  def new_field_slip_tab
+    [:field_slip_new.t, new_field_slip_path, { class: tab_id(__method__.to_s) }]
+  end
+
+  def show_field_slip_tab(field_slip)
+    [:field_slip_show.t, field_slip_path(field_slip),
+     { class: tab_id(__method__.to_s) }]
+  end
+
+  def edit_field_slip_tab(field_slip)
+    [:field_slip_edit.t, edit_field_slip_path(field_slip),
+     { class: tab_id(__method__.to_s) }]
+  end
+
+  def destroy_field_slip_tab(field_slip)
+    [nil, field_slip, { button: :destroy }]
+  end
```

Comment:
@mo-nathan I've started putting these in separate helper files under "app/helpers/tabs" just so i can find them quickly. 

It may seem like silly file splintering, but it really helped me when i was working on refactoring the right-hand tabs, being able to scan the tabs in all the different controllers, in order to spot cases that needed special handling.

Reply:
I don't love having multiple files with the same name (e.g., app/helpers/images.rb and app/helpers/tabs/images.rb), but I've conformed to this convention.

Reply:
Thanks. I didn't love it at first either, and tried to think of alternates, but then I realized how many files we have called `edit.html.erb`, `show.html.erb` etc that are only really identified by the full pathname. 

I realize it may feel like less of an issue for me in VS Code because that editor always displays the pathname in the tab of the file, whereas other editors may not do that.

![Screen Shot 2024-06-10 at 2 52 36 PM](https://github.com/MushroomObserver/mushroom-observer/assets/1948095/5b83cfa9-8e28-4ec1-92bd-cede2c8d8b6e)

Reply:
To me it is much less of an issue with ERB files than with Ruby files.  I have seen circumstances where the Ruby autoloader loads the wrong file when there is more than one class with the same name.  It's actually disturbing to me that I don't have to mention `Tabs` anywhere for this code to just get loaded.  What should really happen is that the helper/tabs/*.rb code should be a set of objects (probably all with the suffix `Tabs` rather than `Helper`).  That would resolve the issue and might make it even easier to manage through a shared base class.

Reply:
Yes - The autoloader/constant name duplication did not occur to me, and that is a better solution.